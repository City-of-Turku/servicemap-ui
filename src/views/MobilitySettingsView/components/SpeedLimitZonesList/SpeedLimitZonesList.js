import React from 'react';
import { Checkbox, FormControlLabel, Typography } from '@material-ui/core';
import PropTypes from 'prop-types';

const SpeedLimitZonesList = ({
  classes, intl, openSpeedLimitList, speedLimitListAsc, speedLimitSelections, setState,
}) => (openSpeedLimitList ? (
  <>
    <div className={`${classes.paragraph} ${classes.border}`}>
      <Typography
        variant="body2"
        aria-label={intl.formatMessage({ id: 'mobilityPlatform.menu.speedLimitZones.select' })}
      >
        {intl.formatMessage({ id: 'mobilityPlatform.menu.speedLimitZones.select' })}
      </Typography>
    </div>
    <div className={classes.buttonList}>
      {openSpeedLimitList
          && speedLimitListAsc.length > 0
          && speedLimitListAsc.map(item => (
            <div key={item} className={classes.checkBoxContainer}>
              <FormControlLabel
                control={(
                  <Checkbox
                    checked={speedLimitSelections.includes(item)}
                    aria-checked={speedLimitSelections.includes(item)}
                    className={classes.margin}
                    onChange={() => setState(item)}
                  />
                )}
                label={(
                  <Typography
                    variant="body2"
                    aria-label={intl.formatMessage(
                      {
                        id: 'mobilityPlatform.content.speedLimitZones.suffix',
                      },
                      { item },
                    )}
                  >
                    {intl.formatMessage(
                      {
                        id: 'mobilityPlatform.content.speedLimitZones.suffix',
                      },
                      { item },
                    )}
                  </Typography>
                )}
              />
            </div>
          ))}
    </div>
  </>
) : null);

SpeedLimitZonesList.propTypes = {
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  openSpeedLimitList: PropTypes.bool,
  speedLimitListAsc: PropTypes.arrayOf(PropTypes.number),
  speedLimitSelections: PropTypes.arrayOf(PropTypes.number),
  setState: PropTypes.func.isRequired,
};

SpeedLimitZonesList.defaultProps = {
  openSpeedLimitList: false,
  speedLimitListAsc: [],
  speedLimitSelections: [],
};

export default SpeedLimitZonesList;
