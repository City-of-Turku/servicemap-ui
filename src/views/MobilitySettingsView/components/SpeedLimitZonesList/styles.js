const styles = theme => ({
  paragraph: {
    textAlign: 'left',
    padding: theme.spacing(1.5),
  },
  border: {
    borderBottom: '1px solid #6f7276',
  },
  margin: {
    marginLeft: theme.spacing(4),
  },
  buttonList: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'left',
  },
  checkBoxContainer: {
    width: '100%',
    borderBottom: '1px solid #6f7276',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'start',
  },
});

export default styles;
