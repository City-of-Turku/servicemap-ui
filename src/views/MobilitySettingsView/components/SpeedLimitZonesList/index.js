import SpeedLimitZonesList from './SpeedLimitZonesList';

export default SpeedLimitZonesList;
