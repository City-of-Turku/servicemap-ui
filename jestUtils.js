import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });