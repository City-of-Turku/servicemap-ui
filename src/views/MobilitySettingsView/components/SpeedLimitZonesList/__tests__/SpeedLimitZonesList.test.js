// Link.react.test.js
import React from 'react';
import SpeedLimitZonesList from '../index';
import { getRenderWithProviders } from '../../../../../../jestUtils';
import finnishTranslations from '../../../../../i18n/fi';

const mockProps = {
  openSpeedLimitList: true,
  speedLimitListAsc: [30, 40, 50],
  speedLimitSelections: [30],
  setState: jest.fn,
};

const renderWithProviders = getRenderWithProviders({});

describe('<SpeedLimitZonesList />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<SpeedLimitZonesList {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<SpeedLimitZonesList {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toContain(finnishTranslations['mobilityPlatform.menu.speedLimitZones.select']);
    expect(p[1].textContent).toContain(`${mockProps.speedLimitListAsc[0]} km/t`);
    expect(p[2].textContent).toContain(`${mockProps.speedLimitListAsc[1]} km/t`);
    expect(p[3].textContent).toContain(`${mockProps.speedLimitListAsc[2]} km/t`);
  });
});
