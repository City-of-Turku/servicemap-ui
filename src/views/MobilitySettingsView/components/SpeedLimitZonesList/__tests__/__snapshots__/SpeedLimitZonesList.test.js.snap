// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpeedLimitZonesList /> should work 1`] = `
<div>
  <div
    class="injectIntl(SpeedLimitZonesList)-paragraph-1 injectIntl(SpeedLimitZonesList)-border-2"
  >
    <p
      aria-label="Voit valintojen avulla näyttää nopeusrajoitusalueita kartalla."
      class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
    >
      Voit valintojen avulla näyttää nopeusrajoitusalueita kartalla.
    </p>
  </div>
  <div
    class="injectIntl(SpeedLimitZonesList)-buttonList-4"
  >
    <div
      class="injectIntl(SpeedLimitZonesList)-checkBoxContainer-5"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          aria-checked="true"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary injectIntl(SpeedLimitZonesList)-margin-3 css-1ov5m8d-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6yqi1c-MuiSvgIcon-root"
            data-testid="CheckBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </span>
        <p
          aria-label="30 km/t"
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          30 km/t
        </p>
      </label>
    </div>
    <div
      class="injectIntl(SpeedLimitZonesList)-checkBoxContainer-5"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          aria-checked="false"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary injectIntl(SpeedLimitZonesList)-margin-3 css-1ov5m8d-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6yqi1c-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <p
          aria-label="40 km/t"
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          40 km/t
        </p>
      </label>
    </div>
    <div
      class="injectIntl(SpeedLimitZonesList)-checkBoxContainer-5"
    >
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          aria-checked="false"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary injectIntl(SpeedLimitZonesList)-margin-3 css-1ov5m8d-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6yqi1c-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <p
          aria-label="50 km/t"
          class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
        >
          50 km/t
        </p>
      </label>
    </div>
  </div>
</div>
`;
